2013-11-25T11:18:41.098138+00:00 heroku[api]: Deploy 2811138 by andygee@gmail.com
2013-11-25T11:18:41.113716+00:00 heroku[api]: Release v21 created by andygee@gmail.com
2013-11-25T11:18:41+00:00 heroku[slug-compiler]: Slug compilation finished
2013-11-25T11:18:41.441329+00:00 heroku[web.1]: State changed from up to starting
2013-11-25T11:18:44.823965+00:00 heroku[web.1]: Stopping all processes with SIGTERM
2013-11-25T11:18:45.573129+00:00 app[web.1]: 2013-11-25 11:18:45 [2] [INFO] Handling signal: term
2013-11-25T11:18:45.573677+00:00 app[web.1]: 2013-11-25 11:18:45 [7] [INFO] Worker exiting (pid: 7)
2013-11-25T11:18:45.707975+00:00 app[web.1]: 2013-11-25 11:18:45 [2] [INFO] Shutting down: Master
2013-11-25T11:18:46.911100+00:00 heroku[web.1]: Starting process with command `gunicorn rosterrun:app`
2013-11-25T11:18:47.396828+00:00 heroku[web.1]: Process exited with status 0
2013-11-25T11:18:48.609395+00:00 app[web.1]: 2013-11-25 11:18:48 [2] [INFO] Starting gunicorn 18.0
2013-11-25T11:18:48.610135+00:00 app[web.1]: 2013-11-25 11:18:48 [2] [INFO] Listening at: http://0.0.0.0:37713 (2)
2013-11-25T11:18:48.610269+00:00 app[web.1]: 2013-11-25 11:18:48 [2] [INFO] Using worker: sync
2013-11-25T11:18:48.618498+00:00 app[web.1]: 2013-11-25 11:18:48 [7] [INFO] Booting worker with pid: 7
2013-11-25T11:18:49.200391+00:00 heroku[web.1]: State changed from starting to up
2013-11-25T11:18:49.933281+00:00 heroku[api]: Starting process with command `python` by andygee@gmail.com
2013-11-25T11:18:55.286206+00:00 heroku[run.7106]: Awaiting client
2013-11-25T11:18:55.333052+00:00 heroku[run.7106]: Starting process with command `python`
2013-11-25T11:18:55.477101+00:00 heroku[run.7106]: State changed from starting to up
2013-11-25T11:19:27.712209+00:00 heroku[run.7106]: Process exited with status 0
2013-11-25T11:19:27.755919+00:00 heroku[run.7106]: State changed from up to complete
2013-11-25T11:19:32.722623+00:00 heroku[router]: at=info method=GET path=/ host=rosterrun.herokuapp.com fwd="67.188.246.249" dyno=web.1 connect=1ms service=35ms status=200 bytes=684
2013-11-25T11:19:40.327958+00:00 app[web.1]: dual client assignment [('Andy', 'SPKiller'), ('Andy', 'Healer'), ('Joe', 'SPActive'), ('Joe', 'Freezer'), ('Cheng', 'Support'), ('Cheng', 'Lurer')]
2013-11-25T11:19:40.655978+00:00 app[web.1]: dual client assignment [('Andy', 'SPKiller'), ('Andy', 'Lurer'), ('Joe', 'SPActive'), ('Joe', 'Freezer'), ('Cheng', 'Support'), ('Cheng', 'Healer')]
2013-11-25T11:19:40.870005+00:00 app[web.1]: dual client assignment [('Andy', 'SPKiller'), ('Andy', 'Support'), ('Joe', 'SPActive'), ('Joe', 'Freezer'), ('Cheng', 'Healer'), ('Cheng', 'Lurer')]
2013-11-25T11:19:43.246962+00:00 app[web.1]: dual client assignment [('Andy', 'Healer'), ('Andy', 'Support')]
2013-11-25T11:19:43.258833+00:00 app[web.1]: dual client assignment [('Andy', 'Healer'), ('Andy', 'Support'), ('Joe', 'Freezer'), ('Joe', 'Healer')]
2013-11-25T11:19:43.259781+00:00 app[web.1]: dual client assignment [('Andy', 'Healer'), ('Andy', 'Support'), ('Cheng', 'Healer'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.286338+00:00 app[web.1]: dual client assignment [('Joe', 'Support'), ('Joe', 'Freezer')]
2013-11-25T11:19:43.288574+00:00 app[web.1]: dual client assignment [('Cheng', 'Support'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.311347+00:00 app[web.1]: dual client assignment [('Joe', 'Support'), ('Joe', 'Freezer'), ('Cheng', 'Healer'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.313649+00:00 app[web.1]: dual client assignment [('Joe', 'Freezer'), ('Joe', 'Healer'), ('Cheng', 'Support'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.428313+00:00 app[web.1]: dual client assignment [('Joe', 'Freezer'), ('Joe', 'Healer')]
2013-11-25T11:19:43.429180+00:00 app[web.1]: dual client assignment [('Cheng', 'Healer'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.456250+00:00 app[web.1]: dual client assignment [('Joe', 'Freezer'), ('Joe', 'Healer'), ('Cheng', 'Healer'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.520542+00:00 app[web.1]: dual client assignment [('Joe', 'Support'), ('Joe', 'Freezer'), ('Cheng', 'Healer'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.522852+00:00 app[web.1]: dual client assignment [('Joe', 'Freezer'), ('Joe', 'Healer'), ('Cheng', 'Support'), ('Cheng', 'Killer')]
2013-11-25T11:19:43.602000+00:00 app[web.1]:     return self.wsgi_app(environ, start_response)
2013-11-25T11:19:43.602000+00:00 app[web.1]: 2013-11-25 11:19:43 [7] [ERROR] Error handling request
2013-11-25T11:19:43.602000+00:00 app[web.1]: Traceback (most recent call last):
2013-11-25T11:19:43.602000+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
2013-11-25T11:19:43.602000+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 131, in handle_request
2013-11-25T11:19:43.602000+00:00 app[web.1]:     respiter = self.wsgi(environ, resp.start_response)
2013-11-25T11:19:43.602000+00:00 app[web.1]:     response = self.make_response(self.handle_exception(e))
2013-11-25T11:19:43.602000+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
2013-11-25T11:19:43.602000+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1403, in handle_exception
2013-11-25T11:19:43.602228+00:00 app[web.1]:     response = self.full_dispatch_request()
2013-11-25T11:19:43.602228+00:00 app[web.1]:     rv = self.handle_user_exception(e)
2013-11-25T11:19:43.602228+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
2013-11-25T11:19:43.602228+00:00 app[web.1]:     reraise(exc_type, exc_value, tb)
2013-11-25T11:19:43.602390+00:00 app[web.1]:     return getattr(self.registry(), name)(*args, **kwargs)
2013-11-25T11:19:43.602000+00:00 app[web.1]:     reraise(exc_type, exc_value, tb)
2013-11-25T11:19:43.602228+00:00 app[web.1]:     rv = self.dispatch_request()
2013-11-25T11:19:43.602228+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
2013-11-25T11:19:43.602228+00:00 app[web.1]:     return self.view_functions[rule.endpoint](**req.view_args)
2013-11-25T11:19:43.602228+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
2013-11-25T11:19:43.602390+00:00 app[web.1]:     db.session.commit()
2013-11-25T11:19:43.602390+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 149, in do
2013-11-25T11:19:43.602390+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 354, in commit
2013-11-25T11:19:43.602835+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 1818, in flush
2013-11-25T11:19:43.602390+00:00 app[web.1]:     self._prepare_impl()
2013-11-25T11:19:43.602390+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 334, in _prepare_impl
2013-11-25T11:19:43.602228+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
2013-11-25T11:19:43.602390+00:00 app[web.1]:   File "/app/rosterrun.py", line 124, in run_calculation
2013-11-25T11:19:43.602390+00:00 app[web.1]:     self.transaction.commit()
2013-11-25T11:19:43.602228+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
2013-11-25T11:19:43.602989+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 525, in execute
2013-11-25T11:19:43.602835+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 1900, in _flush
2013-11-25T11:19:43.602835+00:00 app[web.1]:     flush_context.execute()
2013-11-25T11:19:43.602835+00:00 app[web.1]:     self._flush(objects)
2013-11-25T11:19:43.602835+00:00 app[web.1]:     transaction.rollback(_capture_exception=True)
2013-11-25T11:19:43.602835+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 1936, in _flush
2013-11-25T11:19:43.602390+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 721, in commit
2013-11-25T11:19:43.602989+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 662, in execute
2013-11-25T11:19:43.602835+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 58, in __exit__
2013-11-25T11:19:43.602835+00:00 app[web.1]:     compat.reraise(exc_type, exc_value, exc_tb)
2013-11-25T11:19:43.602989+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 64, in save_obj
2013-11-25T11:19:43.602989+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 569, in _emit_insert_statements
2013-11-25T11:19:43.602835+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 372, in execute
2013-11-25T11:19:43.602989+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 761, in _execute_clauseelement
2013-11-25T11:19:43.602835+00:00 app[web.1]:     rec.execute(self)
2013-11-25T11:19:43.602989+00:00 app[web.1]:     uow
2013-11-25T11:19:43.602390+00:00 app[web.1]:     self.session.flush()
2013-11-25T11:19:43.603292+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 874, in _execute_context
2013-11-25T11:19:43.603292+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 196, in raise_from_cause
2013-11-25T11:19:43.603292+00:00 app[web.1]:     reraise(type(exception), exception, tb=exc_tb)
2013-11-25T11:19:43.603292+00:00 app[web.1]:     context)
2013-11-25T11:19:43.602989+00:00 app[web.1]:     params)
2013-11-25T11:19:43.603292+00:00 app[web.1]:     context)
2013-11-25T11:19:43.602989+00:00 app[web.1]:     compiled_sql, distilled_params
2013-11-25T11:19:43.603292+00:00 app[web.1]:     cursor.execute(statement, parameters)
2013-11-25T11:19:43.603460+00:00 app[web.1]: IntegrityError: (IntegrityError) duplicate key value violates unique constraint "party_combination_g_spreadsheet_id_key"
2013-11-25T11:19:43.603292+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 324, in do_execute
2013-11-25T11:19:43.603460+00:00 app[web.1]: DETAIL:  Key (g_spreadsheet_id)=(t0r5xf97B-k70k6TxherGcw) already exists.
2013-11-25T11:19:43.602989+00:00 app[web.1]:     table, insert)
2013-11-25T11:19:43.602989+00:00 app[web.1]:     execute(statement, params)
2013-11-25T11:19:43.603292+00:00 app[web.1]:     exc_info
2013-11-25T11:19:43.603292+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 867, in _execute_context
2013-11-25T11:19:43.603292+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1024, in _handle_dbapi_exception
2013-11-25T11:19:43.603460+00:00 app[web.1]:  'INSERT INTO party_combination (g_spreadsheet_id, g_worksheet_id, "partyIndex", "instanceName", "playerName", name, "className", rolename) VALUES (%(g_spreadsheet_id)s, %(g_worksheet_id)s, %(partyIndex)s, %(instanceName)s, %(playerName)s, %(name)s, %(className)s, %(rolename)s) RETURNING party_combination.id' {'name': 'ctrlx', 'partyIndex': '0', 'playerName': 'Andy', 'className': 'Clown', 'g_spreadsheet_id': 't0r5xf97B-k70k6TxherGcw', 'rolename': 'Support', 'g_worksheet_id': 'od6', 'instanceName': 'Niddhogg'}
2013-11-25T11:19:43.645718+00:00 heroku[router]: at=info method=POST path=/runcalc host=rosterrun.herokuapp.com fwd="67.188.246.249" dyno=web.1 connect=0ms service=5081ms status=500 bytes=135
